### PR TITLE
RavenDB-15586 fixing leak in EncryptionBuffersPool that could occur when TryPush fails in per core container

### PR DESCRIPTION
**Pull request by ppekrol**

### CODE REVIEW

File: src/Voron/Impl/EncryptionBuffersPool.cs [hunk 1]
```diff
@@ -66,6 +67,9 @@ public EncryptionBuffersPool()
 
                 thread = NativeMemory.ThreadAllocations.Value;
                 thread.Allocations += size;
+
+                Debug.Assert(size == allocation.Size, "size == allocation.Size");
```
Comment (grisha-kotler) [inline src/Voron/Impl/EncryptionBuffersPool.cs]: Can you add the `size` and the `allocation.Size`?

**Review by ayende — APPROVED** [review]

**Comment by grisha-kotler** [comment]
This is needed in 5.0 as well

File: src/Sparrow/Json/JsonOperationContext.cs [hunk 2]
```diff
@@ -973,7 +978,12 @@ protected internal virtual unsafe void Reset(bool forceReleaseLongLivedAllocator
             for (var i = 0; i < _numberOfAllocatedStringsValues; i++)
                 _allocateStringValues[i].Reset();
 
-            _perCoreLazyStringValuesList.Push(_allocateStringValues);
+            if (_perCoreLazyStringValuesList.TryPush(_allocateStringValues) == false)
```
Comment (ayende) [inline src/Sparrow/Json/JsonOperationContext.cs]: Need to check that this isn't already null. 
Pretty sure that this is why we are getting test failures.

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Sparrow/Json/JsonOperationContext.cs [hunk 3]
```diff
@@ -112,7 +112,12 @@ public JsonOperationContext(int initialSize, int longLivedSize, int maxNumberOfA
                 TryDispose(_arenaAllocatorForLongLivedValues);
                 if (_allocateStringValues != null)
                 {
-                    _perCoreLazyStringValuesList.Push(_allocateStringValues);
+                    if (_perCoreLazyStringValuesList.TryPush(_allocateStringValues) == false)
```
Comment (ayende) [inline src/Sparrow/Json/JsonOperationContext.cs:115]:
```suggestion
                    if (_allocateStringValues != null && _perCoreLazyStringValuesList.TryPush(_allocateStringValues) == false)
```

File: src/Sparrow/Json/JsonOperationContext.cs (hunk 2 above)
Comment (ayende) [inline src/Sparrow/Json/JsonOperationContext.cs]:
```suggestion
            if (_allocateStringValues != null && _perCoreLazyStringValuesList.TryPush(_allocateStringValues) == false)
```

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Sparrow/Json/JsonOperationContext.cs (hunk 3 above)
Comment (ppekrol) [reply src/Sparrow/Json/JsonOperationContext.cs:115]: This is checked one line up?

File: src/Sparrow/Json/JsonOperationContext.cs (hunk 2 above)
Comment (ppekrol) [reply src/Sparrow/Json/JsonOperationContext.cs]: Done

**Review by arekpalinski — APPROVED** [review]

**Pull request merged**